### PR TITLE
Disable hostkey check for ssh

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
https://github.com/rancher/fleet/issues/118

**Comment by StrongMonkey** [comment]
https://github.com/rancher/fleet/issues/93

File: pkg/git/git.go [hunk 1]
```diff
@@ -328,6 +329,20 @@ func (g *Git) gitCmd(output io.Writer, args ...string) error {
...
 	}
 
+	if len(g.knownHosts) != 0 {
+		f, err := ioutil.TempFile("", "known_hosts")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(f.Name())
+		defer f.Close()
+		if _, err := f.Write(g.knownHosts); err != nil {
```
Comment (ibuildthecloud) [inline pkg/git/git.go:339]: Don't you need to close after to the write to ensure everything is flushed to disk?

File: pkg/git/git.go [hunk 2]
```diff
@@ -379,7 +394,7 @@ func (g *Git) injectAgent(cmd *exec.Cmd) (io.Closer, error) {
 		return nil, err
 	}
 
-	cmd.Env = append(cmd.Env, "SSH_AUTH_SOCK="+addr.Name)
+	cmd.Env = append(cmd.Env, "SSH_AUTH_SOCK="+addr.Name, "GIT_SSH_COMMAND="+"ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no")
```
Comment (ibuildthecloud) [inline pkg/git/git.go:397]: Won't this conflict if agent is used?

**Pull request merged**